plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -161,11 +170,15 @@ func (p *plugin) runHelmCommand(
 }
 
 // createNewMergedValuesFile replaces/merges original values file with ValuesInline.
-func (p *plugin) createNewMergedValuesFile() (
-	path string, err error) {
+func (p *plugin) createNewMergedValuesFiles(path string) (
```

Comment:
Sorry for the basic questions, but I'm confused about what's going on here. Is this right?
- Before we had two user inputs for values: ValuesInline (which does not have a helm flag) and ValuesFile (which does have a helm flag). To get the former to Helm, we needed to create one consolidated file. To do that, we needed to know how to resolve conflicts, so ValuesMerge was introduced to configure that.
- Now, we still have ValuesInline, but we also have multiple ValuesFiles, which Helm is capable of accepting individually. It seems to me we're doing a merge on each of them, that would progressively include more of the data, and then passing multiple files to Helm?

Questions:
- What is actually in each file we pass to Helm? Am I right to think that given source files A and B + ValuesInline, the file for A would have A<<ValuesInline, the file for B would have A<<ValuesInline<<B<<ValuesInline (where << means merge). Should we have unit tests on the file merge functions?
- Why didn't we originally just put the ValuesInline into an additional file to pass to Helm?
- The ValuesMerge docstring implies that Kustomize really should be doing the full merge before passing to Helm. In which case I'd expect us to give Helm just one file. But I'm wondering if that is not going to behave as users expect in some way (i.e. if they expect ValuesFiles to be a direct equivalent to multiple -f flags on Helm). Is the plural field incompatible with ValuesMerge, then?

Reply:
As discussed offline, we will address this by using `additionalFiles` to specify additional files, and leave the original `valuesFile` untouched.